Framework/Algorithms/src/FindPeaks.cpp

```diff
@@ -462,8 +464,12 @@ void FindPeaks::findPeaksUsingMariscotti() {
   const int blocksize = static_cast<int>(smoothedData->blocksize());
 
   for (int k = start; k < end; ++k) {
-    const auto &S = smoothedData->y(k);
-    const auto &F = smoothedData->e(k);
+    int ws_i(k);
+    if (singleSpectrum)
+      ws_i = 0;
```

Comment:
Why do you need this? You are just undoing what the computation of `start` and `end` above did. Can't you just get rid of `start` and `end` and simply loop over all spectra in the workspace here?

Reply:
I need to check!